### PR TITLE
[ADD] porting of account_invoice_sequential_dates on 8.0

### PR DESCRIPTION
**Pull request by OpenCode**

### CODE REVIEW

File: account_invoice_sequential_dates/invoice.py [hunk 1]
```diff
@@ -19,11 +19,8 @@
 #
 ##############################################################################
 
-import netsvc
-import pooler, tools
-
-from openerp.osv import fields, orm
-from tools.translate import _
+from openerp.osv import orm
```
Comment (andrea4ever) [inline account_invoice_sequential_dates/invoice.py]: Can you replace with odoo standard import

from openerp import models, _

**Comment by andrea4ever** [comment]
replace headers and license with standard OCA for info's 

https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py#L1-L3


Also add some test, if is possible.

Thanks

File: account_invoice_sequential_dates/AUTHORS.txt [hunk 2]
```diff
@@ -6,4 +6,4 @@ Mauro Soligo <mauro.soligo@katodo.com>
 Giovanni Barzan <giovanni.barzan@gmail.com>
 Lorenzo Battistini <lorenzo.battistini@albatos.com>
 Roberto Onnis <onnis.roberto@gmail.com>
-
+Francesco Apruzzese <f.apruzzese@apuliasoftware.it>
```
Comment (tafaRU) [inline account_invoice_sequential_dates/AUTHORS.txt]: Could you please extract to README?

File: account_invoice_sequential_dates/__openerp__.py [hunk 3]
```diff
@@ -1,42 +1,17 @@
...
-Functionalities:
-
-- Check invoice date consistency
-
-""",
     'author': "OpenERP Italian Community,Odoo Community Association (OCA)",
     'website': 'http://www.openerp-italia.org',
     'license': 'AGPL-3',
     "depends" : ['account',],
     "data" : [],
```
Comment (tafaRU) [inline account_invoice_sequential_dates/__openerp__.py]: Please remove empty key

File: account_invoice_sequential_dates/__openerp__.py [hunk 4]
```diff
@@ -1,42 +1,17 @@
...
     "demo_xml" : [],
```
Comment (tafaRU) [inline account_invoice_sequential_dates/__openerp__.py]: Please remove this key as obsolete.

File: account_invoice_sequential_dates/README.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+================================
+
+This module customizes Odoo in order to make invoices with consistent dates
+
+Usage
+=====
+
+When an user validate an invoice, Odoo reads data and number and checks if exist an invoice with bigger data, yet
+
+#. Go to ...
```
Comment (tafaRU) [inline account_invoice_sequential_dates/README.rst]: Please remove this line as it's useless.

File: account_invoice_sequential_dates/README.rst [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+Credits
+=======
+
+Images
+------
+
+* Odoo Community Association: `Icon <https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.svg>`_.
+
+Contributors
```
Comment (tafaRU) [inline account_invoice_sequential_dates/README.rst]: Please move this section under Credits.

File: account_invoice_sequential_dates/__openerp__.py [hunk 7]
```diff
@@ -1,42 +1,18 @@
...
     'license': 'AGPL-3',
-    "depends" : ['account',],
-    "data" : [],
-    "demo_xml" : [],
-    "active": False,
-    'installable': False
+    "depends": [
+        'account',
+        ],
+    "data": [],
```
Comment (tafaRU) [inline account_invoice_sequential_dates/__openerp__.py]: Please remove empty keys

File: account_invoice_sequential_dates/invoice.py [hunk 8]
```diff
@@ -1,47 +1,30 @@
...
 
-from openerp.osv import fields, orm
-from tools.translate import _
 
-class account_invoice(orm.Model):
+class AccountInvoice(models.Model):
+
     _inherit = 'account.invoice'
 
     def action_number(self, cr, uid, ids, context=None):
```
Comment (tafaRU) [inline account_invoice_sequential_dates/invoice.py]: It should be better to use the new API, could you please modify it?

File: account_invoice_sequential_dates/invoice.py [hunk 9]
```diff
@@ -1,47 +1,29 @@
...
-            res = self.search(cr, uid, [('type','=',inv_type),('date_invoice','>',date_invoice), 
-                ('number', '<', number), ('journal_id','=',journal)], context=context)
-            if res:
-                raise orm.except_orm(_('Date Inconsistency'),
-                        _('Cannot create invoice! Post the invoice with a greater date'))
-        return True
+    _inherit = 'account.invoice'
 
-# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
+    @api.model
```
Comment (tafaRU) [inline account_invoice_sequential_dates/invoice.py]: api.multi is the right decorator!

Comment (OpenCode) [inline account_invoice_sequential_dates/invoice.py]: @taraRU done!

File: account_invoice_sequential_dates/README.rst (hunk 5 above)
Comment (OpenCode) [inline account_invoice_sequential_dates/README.rst]: @tafaRU done

**Comment by OpenCode** [comment]
@tafaRU @andrea4ever @lepistone @eLBati All green! Now all testes are correct to be compatible with account_invoice_sequential_dates checks

File: account_invoice_sequential_dates/__openerp__.py [hunk 10]
```diff
@@ -1,42 +1,16 @@
...
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2016 Apulia Software srl <info@apuliasoftware.it>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
 {
     'name': 'Check invoice date consistency',
-    'version': '0.1',
+    'version': '8.0.0.1.0',
```
Comment (tafaRU) [inline account_invoice_sequential_dates/__openerp__.py]: It should be 8.0.1.0.0, please fix it.

File: account_vat_period_end_statement/tests/test_vat_statement.py [hunk 11]
```diff
@@ -23,6 +23,12 @@ def setUp(self):
         self.current_period = self.period_model.find()
         self.vat_statement_model = self.env['account.vat.period.end.statement']
 
+        # ----- Set invoice date to biggest date in the system
```
Comment (tafaRU) [inline account_vat_period_end_statement/tests/test_vat_statement.py]: I would replace 'biggest date' by youngest date'.

File: account_vat_period_end_statement/tests/test_vat_statement.py [hunk 12]
```diff
@@ -23,6 +23,12 @@ def setUp(self):
...
+        # ----- This solves problems with account_invoice_sequential_dates
+        self.biggest_data = self.invoice_model.search(
```
Comment (tafaRU) [inline account_vat_period_end_statement/tests/test_vat_statement.py]: I would replace 'biggest_data' by 'youngest_date'

File: account_vat_period_end_statement/tests/test_vat_statement.py [hunk 13]
```diff
@@ -75,6 +81,7 @@ def setUp(self):
 
     def test_vat_statement(self):
         out_invoice = self.invoice_model.create({
+            'date_invoice': self.biggest_data,
```
Comment (tafaRU) [inline account_vat_period_end_statement/tests/test_vat_statement.py]: Same as above

File: account_vat_period_end_statement/tests/test_vat_statement.py [hunk 14]
```diff
@@ -91,6 +98,7 @@ def test_vat_statement(self):
         out_invoice.signal_workflow('invoice_open')
 
         in_invoice = self.invoice_model.create({
+            'date_invoice': self.biggest_data,
```
Comment (tafaRU) [inline account_vat_period_end_statement/tests/test_vat_statement.py]: Same as above

File: l10n_it_ricevute_bancarie/test/riba_invoice.yml [hunk 15]
```diff
@@ -16,6 +16,19 @@
     partner_id: base.res_partner_12
     payment_term: account_payment_term_riba
 -
+  Set invoice date to biggest date in the system
```
Comment (tafaRU) [inline l10n_it_ricevute_bancarie/test/riba_invoice.yml]: Same as above.

File: l10n_it_ricevute_bancarie/test/riba_invoice.yml [hunk 16]
```diff
@@ -16,6 +16,19 @@
...
+  This solves problems with account_invoice_sequential_dates
+-
+  !python {model: account.invoice}: |
+    invoice_ids = self.search(cr, uid, [('date_invoice', '!=', False)],
+                              order='date_invoice desc', limit=1,
+                              context=context)
+    if invoice_ids:
+      biggest_invoice = self.browse(cr, uid, invoice_ids[0], context=context)
```
Comment (tafaRU) [inline l10n_it_ricevute_bancarie/test/riba_invoice.yml]: s/biggest_invoice/youngest_invoice

File: l10n_it_ricevute_bancarie/test/unsolved_riba.yml [hunk 17]
```diff
@@ -16,6 +16,19 @@
     partner_id: base.res_partner_12
     payment_term: account_payment_term_riba
 -
+  Set invoice date to biggest date in the system
```
Comment (tafaRU) [inline l10n_it_ricevute_bancarie/test/unsolved_riba.yml]: Same as above.

File: l10n_it_ricevute_bancarie/test/unsolved_riba.yml [hunk 18]
```diff
@@ -16,6 +16,19 @@
...
+  This solves problems with account_invoice_sequential_dates
+-
+  !python {model: account.invoice}: |
+    invoice_ids = self.search(cr, uid, [('date_invoice', '!=', False)],
+                              order='date_invoice desc', limit=1,
+                              context=context)
+    if invoice_ids:
+      biggest_invoice = self.browse(cr, uid, invoice_ids[0], context=context)
```
Comment (tafaRU) [inline l10n_it_ricevute_bancarie/test/unsolved_riba.yml]: Same as above.

**Comment by tafaRU** [comment]
@OpenCode, thanks a lot!
Last fixes to tests LGTM, I'd just replace biggest by youngest everywhere it appears, since the second one seems more  suitable.

**Comment by eLBati** [comment]
@tafaRU @OpenCode if you want to change it, I would vote for the term ```recent``` (most recent) then

**Comment by OpenCode** [comment]
@tafaRU @eLBati maybe "recent_date" is better

**Comment by tafaRU** [comment]
> "recent_date" is better

+1

**Comment by OpenCode** [comment]
@tafaRU @eLBati @andrea4ever Travis is red because link used by geonames is broken but all testes changed by my last commit are ok

**Comment by tafaRU** [comment]
@OpenCode, see https://github.com/OCA/partner-contact/pull/268

**Comment by OpenCode** [comment]
@tafaRU Have we to wait to merge?

**Comment by tafaRU** [comment]
@OpenCode, I restarted the build and now it looks :green_heart:

So :+1:

**Comment by OpenCode** [comment]
@tafaRU I <3 you

**Comment by tafaRU** [comment]
> @tafaRU I <3 you

To prove that you could add me as contributor :wink:

**Comment by OpenCode** [comment]
@tafaRU done

**Comment by andrea4ever** [comment]
:+1: For me! Nice Work

**Comment by OpenCode** [comment]
@andrea4ever thanks :)

@tafaRU @eLBati can we merge? Do you need other informations?

**Comment by scigghia** [comment]
:+1:

**Pull request merged**